<!-- 
RSS generated by JIRA (5.2.8#851-sha1:3262fdc28b4bc8b23784e13eadc26a22399f5d88) at Mon Jul 08 06:17:24 UTC 2013

It is possible to restrict the fields that are returned in this document by specifying the 'field' parameter in your request.
For example, to request only the issue key and summary add field=key&field=summary to the URL of your request.
For example:
https://issues.apache.org/jira/si/jira.issueviews:issue-xml/HTTPCLIENT-436/HTTPCLIENT-436.xml?field=key&field=summary
-->
<rss version="0.92" >
<channel>
    <title>ASF JIRA</title>
    <link>https://issues.apache.org/jira</link>
    <description>This file is an XML representation of an issue</description>
    <language>en-uk</language>    <build-info>
        <version>5.2.8</version>
        <build-number>851</build-number>
        <build-date>26-02-2013</build-date>
    </build-info>

<item>
            <title>[HTTPCLIENT-436] Retry on ConnectionException does not work</title>
                <link>https://issues.apache.org/jira/browse/HTTPCLIENT-436</link>
                <project id="12310360" key="HTTPCLIENT">HttpComponents HttpClient</project>
                        <description>I noticed that the Retry handler mechanism does not work when the client cannot&lt;br/&gt;
initiate a connection (which throws java.net.ConnectionException). This happens&lt;br/&gt;
for me for instance when there is proxy and a tunneling in the picture and&lt;br/&gt;
sometimes there are connectivity problems.&lt;br/&gt;
&lt;br/&gt;
I had my own RetryHandler, however, the Connection Timeout exception never falls&lt;br/&gt;
in it. I took a look at the source code and noticed that the open() method and&lt;br/&gt;
any thrown exception at this level occurs outside the control of the Retry&lt;br/&gt;
Handler (which seems to be involved only after open() succeeds).&lt;br/&gt;
&lt;br/&gt;
In fact, if the open() throws ConnectionException (as is my case), since the&lt;br/&gt;
try/catch wrapping the open() is not inside the while() but on top of it, it&lt;br/&gt;
stops the loop and the retry handler does not get a chance to be invoked .&lt;br/&gt;
&lt;br/&gt;
riad</description>
                <environment>Operating System: Windows XP&lt;br/&gt;
Platform: PC</environment>
            <key id="12333995">HTTPCLIENT-436</key>
            <summary>Retry on ConnectionException does not work</summary>
                <type id="1" iconUrl="https://issues.apache.org/jira/images/icons/issuetypes/bug.png">Bug</type>
                                <priority id="3" iconUrl="https://issues.apache.org/jira/images/icons/priorities/major.png">Major</priority>
                    <status id="6" iconUrl="https://issues.apache.org/jira/images/icons/statuses/closed.png">Closed</status>
                    <resolution id="1">Fixed</resolution>
                                <assignee username="-1">Unassigned</assignee>
                                <reporter username="rsouissi.c@stc.com.sa">riad</reporter>
                        <labels>
                    </labels>
                <created>Tue, 22 Feb 2005 04:01:44 +0000</created>
                <updated>Wed, 16 Feb 2011 20:44:45 +0000</updated>
                    <resolved>Sun, 22 Apr 2007 08:10:58 +0100</resolved>
                            <version>3.0 Final</version>
                                <fixVersion>3.0 Final</fixVersion>
                                <component>HttpClient</component>
                        <due></due>
                    <votes>0</votes>
                        <watches>0</watches>
                                                    <comments>
                    <comment id="12382169" author="olegk" created="Tue, 22 Feb 2005 20:09:15 +0000"  >Created an attachment (id=14341)&lt;br/&gt;
Patch (take 1)&lt;br/&gt;
&lt;br/&gt;
This should take care of the problem. Riad, could you please test the patch&lt;br/&gt;
against the SVN trunk in your environment?&lt;br/&gt;
&lt;br/&gt;
Oleg</comment>
                    <comment id="12382170" author="rsouissi.c@stc.com.sa" created="Tue, 22 Feb 2005 23:46:38 +0000"  >It seems to work. It retries on ConnectionException and my RetryHandler is &lt;br/&gt;
invoked.&lt;br/&gt;
Thanks for the fix.&lt;br/&gt;
&lt;br/&gt;
riad</comment>
                    <comment id="12382171" author="olegk" created="Tue, 22 Mar 2005 04:23:07 +0000"  >Any objections to committing this patch?&lt;br/&gt;
&lt;br/&gt;
Oleg</comment>
                    <comment id="12382172" author="becke@u.washington.edu" created="Wed, 23 Mar 2005 02:15:42 +0000"  >Looks good to me.&lt;br/&gt;
&lt;br/&gt;
Mike</comment>
                    <comment id="12382173" author="olegk" created="Wed, 30 Mar 2005 08:34:04 +0100"  >Patch committed.&lt;br/&gt;
&lt;br/&gt;
Oleg</comment>
                </comments>
                    <attachments>
                    <attachment id="12327202" name="ASF.LICENSE.NOT.GRANTED--connect.patch" size="6369" author="olegk" created="Tue, 22 Feb 2005 20:09:15 +0000" />
                </attachments>
            <subtasks>
        </subtasks>
                <customfields>
                                <customfield id="customfield_12310310" key="com.atlassian.jira.toolkit:attachments">
                <customfieldname>Attachment count</customfieldname>
                <customfieldvalues>
                    <customfieldvalue>1.0</customfieldvalue>
                </customfieldvalues>
            </customfield>
                                            <customfield id="customfield_10010" key="com.atlassian.jira.plugin.system.customfieldtypes:importid">
                <customfieldname>Bugzilla Id</customfieldname>
                <customfieldvalues>
                    <customfieldvalue>33677</customfieldvalue>
                </customfieldvalues>
            </customfield>
                                            <customfield id="customfield_12310220" key="com.atlassian.jira.ext.charting:firstresponsedate">
                <customfieldname>Date of First Response</customfieldname>
                <customfieldvalues>
                    <customfieldvalue>Tue, 22 Feb 2005 20:09:15 +0000</customfieldvalue>

                </customfieldvalues>
            </customfield>
                                                                                                        <customfield id="customfield_12310420" key="com.pyxis.greenhopper.jira:gh-global-rank">
                <customfieldname>Global Rank</customfieldname>
                <customfieldvalues>
                    <customfieldvalue>133483</customfieldvalue>
                </customfieldvalues>
            </customfield>
                                            <customfield id="customfield_12310920" key="com.pyxis.greenhopper.jira:gh-global-rank">
                <customfieldname>Rank</customfieldname>
                <customfieldvalues>
                    <customfieldvalue>47127</customfieldvalue>
                </customfieldvalues>
            </customfield>
                                                                                    <customfield id="customfield_12310222" key="com.atlassian.jira.ext.charting:timeinstatus">
                <customfieldname>Time in Status</customfieldname>
                <customfieldvalues>
                    
                </customfieldvalues>
            </customfield>
                            </customfields>
    </item>
</channel>
</rss>